app/controllers/application_controller.rb

```diff
@@ -60,4 +60,11 @@ def basecamp_auth_check
       @basecamp_agent = current_user.agents.where(type: 'Agents::BasecampAgent').first
     end
   end
+
+  def agent_params
+    params.permit!
```

Comment:
The problem is the `options` key, strong parameters does not seem have a way to allow everything inside a nested key. Since the options hash is dynamic we would need to traverse through the `params[:options]` and grab every key.